src/makeInfinityAdder.js

```diff
@@ -6,7 +6,21 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  return function adder(...args) {
+    if (args.length === 0) {
+      const currentSum = sum;
+
+      sum = 0;
+
+      return currentSum;
+    } else {
+      sum += args.reduce((acc, val) => acc + val, 0);
+
+      return adder;
+    }
```

Comment:
no needs to use `else` if you use return earlier
and you can simplify avoiding reduce method
```suggestion
    }
    
    sum += number;

    return adder;
```